crates/anvil/tests/it/pubsub.rs

```diff
@@ -1,275 +1,278 @@
 
     // send a tx triggering an event
     let _receipt = contract
-        .set_value("Next Message".to_string())
+        .setValue("Next Message".to_string())
         .send()
         .await
         .unwrap()
+        .get_receipt()
         .await
-        .unwrap()
         .unwrap();
 
+    let mut log = stream.into_stream();
     // get the emitted event
-    let log = stream.next().await.unwrap().unwrap();
-    assert_eq!(
-        log,
-        ValueChangedFilter {
-            author: from,
-            old_value: "First Message".to_string(),
-            new_value: "Next Message".to_string(),
-        },
-    );
+    let (value_changed, _log) = log.next().await.unwrap().unwrap();
+
+    assert_eq!(value_changed.author, from);
+    assert_eq!(value_changed.oldValue, "First Message".to_string());
+    assert_eq!(value_changed.newValue, "Next Message".to_string());
 }
 
 #[tokio::test(flavor = "multi_thread")]
 async fn test_subscriptions() {
     let (_api, handle) =
         spawn(NodeConfig::test().with_blocktime(Some(std::time::Duration::from_secs(1)))).await;
-    let ws = Ws::connect(handle.ws_endpoint()).await.unwrap();
-
-    // Subscribing requires sending the sub request and then subscribing to
-    // the returned sub_id
-    let sub_id: U256 = ws.request("eth_subscribe", ["newHeads"]).await.unwrap();
-    let mut stream = ws.subscribe(sub_id).unwrap();
-
-    let mut blocks = Vec::new();
-    for _ in 0..3 {
-        let item = stream.next().await.unwrap();
-        let block: Block<TxHash> = serde_json::from_str(item.get()).unwrap();
-        blocks.push(block.number.unwrap_or_default().as_u64());
-    }
+    let provider = connect_pubsub(&handle.ws_endpoint()).await;
+    let sub_id: U256 = provider.raw_request("eth_subscribe".into(), ["newHeads"]).await.unwrap();
+    let stream: Subscription<AlloyBlock> = provider.get_subscription(sub_id).await.unwrap();
+    let blocks = stream
+        .into_stream()
+        .take(3)
+        .collect::<Vec<_>>()
+        .await
+        .into_iter()
+        .map(|b| b.header.number.unwrap())
+        .collect::<Vec<_>>();
 
     assert_eq!(blocks, vec![1, 2, 3])
 }
 
+// TODO: Fix this, num > 17 breaks the test due to poller channel_size defaults to 16. Recv channel
+// lags behind.
 #[tokio::test(flavor = "multi_thread")]
 async fn test_sub_new_heads_fast() {
     let (api, handle) = spawn(NodeConfig::test()).await;
 
-    let provider = ethers_ws_provider(&handle.ws_endpoint());
+    let provider = connect_pubsub(&handle.ws_endpoint()).await;
 
     let blocks = provider.subscribe_blocks().await.unwrap();
+    let mut blocks = blocks.into_stream();
+
+    let num = 1000u64;
+
+    let mut block_numbers = Vec::new();
+    for _ in 0..num {
+        let mine = api.mine_one();
+        let block = async {
+            let block = blocks.next().await.unwrap();
+            block.header.number.unwrap()
+        };
```

Comment:
nit: the async scope is not needed here and can just be extracted and put after the mine.await